packages/editor/src/components/global-styles-renderer/index.js

```diff
@@ -8,10 +8,16 @@ import { useSelect, useDispatch } from '@wordpress/data';
  * Internal dependencies
  */
 import { store as editorStore } from '../../store';
-import { useGlobalStylesOutput } from '../../hooks/use-global-styles-output';
+import { useGlobalStylesOutputWithConfig } from '../../hooks/use-global-styles-output';
+import { useGlobalStylesContext } from '../global-styles-provider';
 
 function useGlobalStylesRenderer( disableRootPadding ) {
-	const [ styles, settings ] = useGlobalStylesOutput( disableRootPadding );
+	// Use useGlobalStylesContext which includes connected style variation for the current post.
+	const { merged: mergedConfig } = useGlobalStylesContext();
+	const [ styles, settings ] = useGlobalStylesOutputWithConfig(
+		mergedConfig,
+		disableRootPadding
+	);
```

Comment:
This probably is going to happen anyway, but it'd good to be able to preview how the styles look in the preview window while creating them. 

If it's already working in this PR, then ignore me! Maybe just my setup.